### ISSUE TITLE
Some int prefs shown as floats, and some float prefs doesn't show value field at all.

### ISSUE THREAD

**Issue by TheXRMonk** [sabresaurus/PlayerPrefsEditor#9]
![image](https://user-images.githubusercontent.com/54309738/150538782-4040ad78-b289-4f9a-a839-801f046b7c49.png)

This looks weird. It doesn't affect functionality - the values are saved and retrieved correctly in the correct format, but it just shows up wrong in the editor.

**Comment by rfadeev** [comment]
Hi,

Could you please provide an example of int pref showing as float? On the screenshot I can see int value shown as int 🙂

Speaking of "0" shown for a float instead of "0.0" you might be expecting. This is default behavior for `EditorGUILayout.FloatField`. You can observe the same thing for game object position in transfer component in the inspector. There "0" is displayed instead of "0.0".

As for pref which does not show value at all, seems like a bug where value type is not supported (see type check [here](https://github.com/sabresaurus/PlayerPrefsEditor/blob/5dc7be3429526a92cee1a2f26c982f48f3d58f52/Editor/PlayerPrefsEditor.cs#L667)). It would be helpful to get it reproduced to fix it. Could you please advise what type was used to write value for `findEventsQuest_Progression` key or how is it written by your code? You can set a breakpoint at the line I linked to check the type.

**Comment by TheXRMonk** [comment]
The _completeStatus is supposed to be an int (bool).
Set like this;
```
public bool Completed
    {
        get => PlayerPrefsBool.WithKey(title + "_completeStatus", completed).Value;
        
        private set
        {
            completed = value;
            if(Application.isPlaying)
                PlayerPrefsBool.WithKey(title + "_completeStatus", completed).Value = completed;
        }
    }
```

Using wrapper;
```
public class PlayerPrefsBool : PlayerPrefsType
	{
		public bool Value
		{
			get => PlayerPrefs.GetInt(Key, DefaultValue ? 1 : 0) == 1;
			set => PlayerPrefs.SetInt(Key, value ? 1 : 0);
		}
		public bool DefaultValue;
		
		public static PlayerPrefsBool WithKey(string key, bool defaultValue = false) => new PlayerPrefsBool(key);

		public PlayerPrefsBool(string key, bool defaultValue = false)
		{
			Key = key;
			DefaultValue = defaultValue;
		}
	}
	[Serializable]
	public abstract class PlayerPrefsType
	{
		public string Key { get; protected set; }
		
		public bool IsSet => PlayerPrefs.HasKey(Key);

		public void Delete() => PlayerPrefs.DeleteKey(Key);
	}


```
Here's the code that writes the progression to player prefs; (using the same wrapper for more typesafe assignments).
```
public string uniqueIdentifier;

public float Progression
    {
        get 
        {
            return PlayerPrefsFloat.WithKey(uniqueIdentifier + "_Progression", progression).Value;
        }
        private set
        {
            progression = value;
            if(Application.isPlaying)
                PlayerPrefsFloat.WithKey(uniqueIdentifier + "_Progression", progression).Value = progression;
        }
    }
```

```
namespace MyBox
{
	[Serializable]
	public class PlayerPrefsFloat : PlayerPrefsType
	{
		public float Value
		{
			get => PlayerPrefs.GetFloat(Key, DefaultValue);
			set => PlayerPrefs.SetFloat(Key, value);
		}
		public float DefaultValue;
		
		
		public static PlayerPrefsFloat WithKey(string key, float defaultValue = 0) => new PlayerPrefsFloat(key, defaultValue);

		public PlayerPrefsFloat(string key, float defaultValue = 0)
		{
			Key = key;
			DefaultValue = defaultValue;
		}
	}
}
```
```

[Serializable]
	public abstract class PlayerPrefsType
	{
		public string Key { get; protected set; }
		
		public bool IsSet => PlayerPrefs.HasKey(Key);

		public void Delete() => PlayerPrefs.DeleteKey(Key);
	}
```

The editorprefs are also screwed;
![image](https://user-images.githubusercontent.com/54309738/151324320-29d98a28-157b-485d-b0e7-dfd0bc14ecaa.png)

**Comment by TheXRMonk** [comment]
Also, how are playerpref bools set? Didn't think those where supported, but apparently editorprefs can be bools?

**Comment by rfadeev** [comment]
I failed to reproduce missing value field using your code. I can see value field drawn for `Completed` and `Progression`. Could you please set breakpoint to this [line](https://github.com/sabresaurus/PlayerPrefsEditor/blob/5dc7be3429526a92cee1a2f26c982f48f3d58f52/Editor/PlayerPrefsEditor.cs#L667) and tell me what is `valueType` for the broken keys? Alternatively, you can add logging as last `else` statement to this [line](https://github.com/sabresaurus/PlayerPrefsEditor/blob/5dc7be3429526a92cee1a2f26c982f48f3d58f52/Editor/PlayerPrefsEditor.cs#L820) and share what is logged.
```csharp
                else
                {
                    Debug.Log($"Failed to draw item, displayKey: {displayKey}, valueType {valueType}");
                }
```

[UnityEngine.PlayerPrefs](https://docs.unity3d.com/ScriptReference/PlayerPrefs.html) indeed do not support bool, but [UnityEditor.EditorPrefs](https://docs.unity3d.com/ScriptReference/EditorPrefs.html) do. PlayerPrefs utilities of this repository support bool for encrypted values via prefix, see `PlayerPrefsUtility.VALUE_BOOL_PREFIX` usage for more info. For non-encrypted values there is no bool support from what I can see.

**Comment by asmund9000** [comment]
The problem with float values is still present.
Unity 2021.3.5f1

![Снимок экрана 2022-07-28 124614](https://user-images.githubusercontent.com/24368653/181429840-f70ed364-7ab5-4c45-81d8-f1b57f56f28c.png)

**Comment by sabresaurus** [comment]
I've reproduced the issue, will work on a fix

![image](https://user-images.githubusercontent.com/17784523/182478119-7c45a14b-adfe-4bd8-8418-98e812670743.png)

**Comment by sabresaurus** [comment]
Fixed in 1.4.1
